From 57cf258dad0d881269151c0c844ad8269fe3985e Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Thu, 30 May 2024 11:48:13 +0200
Subject: [PATCH 610/954] feat(fdts): add fmc node for stm32mp21

Add fmc node for stm32mp21.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git a638f1a7052857cae27439fee1d1d092e259fe48 feat(fdts): add fmc node for stm32mp21]
Change-Id: Ibb0b051a5ec248a2f4bf3e8c6e4a10db6e5599bb
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/384301
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp21-bl2.dtsi  |  3 +++
 fdts/stm32mp21-bl31.dtsi |  2 ++
 fdts/stm32mp211.dtsi     | 34 ++++++++++++++++++++++++++++++++++
 3 files changed, 39 insertions(+)

diff --git a/fdts/stm32mp21-bl2.dtsi b/fdts/stm32mp21-bl2.dtsi
index 82d56ec9f8..d62426e614 100644
--- a/fdts/stm32mp21-bl2.dtsi
+++ b/fdts/stm32mp21-bl2.dtsi
@@ -7,6 +7,9 @@
 
 / {
 	soc@0 {
+#if !STM32MP_RAW_NAND
+		/delete-node/ memory-controller@48200000;
+#endif
 #if !STM32MP_EMMC && !STM32MP_SDMMC
 		bus@42080000 {
 			/delete-node/ mmc@48220000;
diff --git a/fdts/stm32mp21-bl31.dtsi b/fdts/stm32mp21-bl31.dtsi
index b54c0e8028..5f45a21d05 100644
--- a/fdts/stm32mp21-bl31.dtsi
+++ b/fdts/stm32mp21-bl31.dtsi
@@ -5,6 +5,8 @@
 
 / {
 	soc@0 {
+		/delete-node/ memory-controller@48200000;
+
 		bus@42080000 {
 			/delete-node/ mmc@48220000;
 			/delete-node/ mmc@48230000;
diff --git a/fdts/stm32mp211.dtsi b/fdts/stm32mp211.dtsi
index a41a848c8c..7a675577b8 100644
--- a/fdts/stm32mp211.dtsi
+++ b/fdts/stm32mp211.dtsi
@@ -388,6 +388,40 @@
 			status = "okay";
 		};
 
+		fmc: memory-controller@48200000 {
+			compatible = "st,stm32mp25-fmc2-ebi";
+			reg = <0x48200000 0x400>;
+			ranges = <0 0 0x70000000 0x04000000>, /* EBI CS 1 */
+				 <1 0 0x74000000 0x04000000>, /* EBI CS 2 */
+				 <2 0 0x78000000 0x04000000>, /* EBI CS 3 */
+				 <3 0 0x7c000000 0x04000000>, /* EBI CS 4 */
+				 <4 0 0x48810000 0x00001000>; /* NAND */
+			#address-cells = <2>;
+			#size-cells = <1>;
+			clocks = <&rcc CK_KER_FMC>;
+			resets = <&rcc FMC_R>;
+			status = "disabled";
+
+			nand-controller@4,0 {
+				compatible = "st,stm32mp25-fmc2-nfc";
+				reg = <4 0x0000 0x10>,
+				      <4 0x0090 0x10>,
+				      <4 0x00a0 0x10>,
+				      <4 0x0400 0x10>,
+				      <4 0x0490 0x10>,
+				      <4 0x04a0 0x10>,
+				      <4 0x0800 0x10>,
+				      <4 0x0890 0x10>,
+				      <4 0x08a0 0x10>,
+				      <4 0x0c00 0x10>,
+				      <4 0x0c90 0x10>,
+				      <4 0x0ca0 0x10>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+				status = "disabled";
+			};
+		};
+
 		pinctrl: pinctrl@44240000 {
 			#address-cells = <1>;
 			#size-cells = <1>;
-- 
2.53.0

